cmd/nexusUpload_test.go

```diff
@@ -617,45 +547,49 @@ func TestUploadMavenProjects(t *testing.T) {
 		utils.setProperty("performance-tests/pom.xml", "project.groupId", "com.mycompany.app")
 		utils.setProperty("performance-tests/pom.xml", "project.artifactId", "my-app-app")
 		utils.setProperty("performance-tests/pom.xml", "project.packaging", "")
-		utils.files["pom.xml"] = testPomXml
-		utils.files["application/pom.xml"] = testPomXml
-		utils.files["application/target/final-artifact.war"] = []byte("contentsOfJar")
-		utils.files["application/target/final-artifact-classes.jar"] = []byte("contentsOfClassesJar")
-		utils.files["integration-tests/pom.xml"] = testPomXml
-		utils.files["integration-tests/target/final-artifact-integration-tests.jar"] = []byte("contentsOfJar")
-		utils.files["unit-tests/pom.xml"] = testPomXml
-		utils.files["unit-tests/target/final-artifact-unit-tests.jar"] = []byte("contentsOfJar")
-		utils.files["performance-tests/pom.xml"] = testPomXml
+		utils.AddFile("pom.xml", testPomXml)
+		utils.AddFile("application/pom.xml", testPomXml)
+		utils.AddFile("application/target/final-artifact.war", []byte("contentsOfJar"))
+		utils.AddFile("application/target/final-artifact-classes.jar", []byte("contentsOfClassesJar"))
+		utils.AddFile("integration-tests/pom.xml", testPomXml)
+		utils.AddFile("integration-tests/target/final-artifact-integration-tests.jar", []byte("contentsOfJar"))
+		utils.AddFile("unit-tests/pom.xml", testPomXml)
+		utils.AddFile("unit-tests/target/final-artifact-unit-tests.jar", []byte("contentsOfJar"))
+		utils.AddFile("performance-tests/pom.xml", testPomXml)
 		uploader := mockUploader{}
 		options := createOptions()
 
 		err := runNexusUpload(&utils, &uploader, &options)
 		assert.NoError(t, err, "expected upload of maven project with application module to succeed")
 		assert.Equal(t, "1.0", uploader.GetArtifactsVersion())
-		assert.Equal(t, "my-app-app", uploader.GetArtifactsID())
+		assert.Equal(t, "my-app", uploader.GetArtifactsID())
 
 		artifacts := uploader.uploadedArtifacts
 		if assert.Equal(t, 4, len(artifacts)) {
-			assert.Equal(t, "pom.xml", artifacts[0].File)
+			assert.Equal(t, "application/pom.xml", artifacts[0].File)
```

Comment:
The order changed. Is it okay that okay?

Reply:
I see, it is because of the sort.

Reply:
I realized that the sorting by length is not actually stable and may still result in flaky tests. Plus it introduced some globally visible stuff (tests in other files in the cmd package were able to re-use this).